math/mobius/mobius.go

```diff
@@ -0,0 +1,44 @@
+// mobius.go
+// description: Returns μ(n)
+// details:
+// For any positive integer n, define μ(n) as the sum of the primitive nth roots of unity.
+// It has values in {−1, 0, 1} depending on the factorization of n into prime factors:
+//   μ(n) = +1 if n is a square-free positive integer with an even number of prime factors.
+//   μ(n) = −1 if n is a square-free positive integer with an odd number of prime factors.
+//   μ(n) = 0 if n has a squared prime factor.
+// wikipedia: https://en.wikipedia.org/wiki/M%C3%B6bius_function
+// author: Akshay Dubey (https://github.com/itsAkshayDubey)
+// see mobius_test.go
+
+package mobius
+
+import (
+	p "github.com/TheAlgorithms/Go/math/prime"
+)
+
+// Mu is the Mobius function
+// This function returns μ(n) for given number
+func Mu(n int) int {
+
+	if n <= 1 {
+		return 1
+	}
+
+	primeFactorCount := 0
+
+	for i := 1; i <= n; i++ {
+		if n%i == 0 && p.OptimizedTrialDivision(int64(i)) {
```

Comment:
Do you know the way to achieve the logical equivalent of `Mu` through Liouville function? I think that might be faster than this one... Though I do remember there was an equivalent step in that way as well so it might not be... (I'm a functional analyst not a number theorist so my statement above might be wrong... 😅)

Reply:
Yes, Mu is the same as the Liouville function with an added check for repeated prime factors. We can create Liouville function and call it in Mu with an additional check for repeated prime factors and in this Liouville function, we can use logic faster than this one. Something like I did in the JS repo as below:
`return number <= 0 ? 'Number must be greater than zero.' : PrimeFactors(number).length !== new Set(PrimeFactors(number)).size ? 0 : PrimeFactors(number).length % 2 === 0 ? 1 : -1`

Reply:
We have primefactorization.go in this repo which returns a map with prime factor as key and occurrence of that factor as value, we can use this in Liouville function and further use this function in Mu.